java/code/src/com/suse/manager/webui/websocket/WebsocketSessionConfigurator.java

```diff
@@ -35,12 +34,12 @@ public void modifyHandshake(ServerEndpointConfig config,
             HandshakeRequest request,
             HandshakeResponse response) {
         HttpSession httpSession = (HttpSession)request.getHttpSession();
-        HttpServletRequest sr = (HttpServletRequest) httpSession.getAttribute("__original_request__");
-        User user = new RequestContext(sr).getCurrentUser();
-        // force roles to be loaded so now since it will fail from the websocket thread
-        if (user != null) {
-            user.getRoles();
-            config.getUserProperties().put("currentUser", user);
-        }
+        Optional.ofNullable(httpSession.getAttribute("webUserID"))
+                .map(webUserIDObject -> UserFactory.lookupById((Long)webUserIDObject))
+                .ifPresent(user -> {
+                    // force roles to be loaded so now since it will fail from the websocket thread
+                    user.getRoles();
```

Comment:
This is the only bit I do not feel excited about. Why are we saving a Hibernated object in the properties, where we know it will be passed to another thread?

Hibernated objects are known not to survive between threads. How do we use this `user` object later? Could we save just the minimally required properties as value types or POJOs to get Hibernate out of the equation?

Reply:
Thanks @moio , that makes sense. I change the code to just save the user Id and then get the user from the database on the websocket handler class. Can you have a look if it's what yous was expecting?